hcloud.tf

```diff
@@ -43,11 +64,86 @@ resource "hcloud_server" "k3s" {
   ssh_keys    = [ data.hcloud_ssh_key.ssh_key.id ]
   location    = "${var.location}"
 
+  public_net {
+    ipv4_enabled = true
+    ipv6_enabled = false
+  }  
+
+  network {
+    network_id = hcloud_network.internal-network.id
+    ip         = "10.0.1.6"
+  }
+
   firewall_ids = [ hcloud_firewall.k3s-firewall.id ]
 
   user_data = "${data.template_file.cloud_init.rendered}"
+
+  labels = {
+    label_1 = "${var.env_name}"
+  }
 }
 
 output "ipv4" {
   value = hcloud_server.k3s.ipv4_address
 }
+
+# certificate
+resource "hcloud_managed_certificate" "lb_cert" {
+  name         = "lb_cert"
+  domain_names = ["*.${var.base_domain}", "${var.base_domain}"]
+  labels = {
+    label_1 = "${var.env_name}"
+  }
+}
+
+# load balancer
+resource "hcloud_load_balancer" "wob-tsp-staging" {
```

Comment:
I'd suggest to make the resource names not use-case specific (see occurances below as well)